stripe/src/main/java/com/stripe/android/IssuingCardPinService.java

```diff
@@ -0,0 +1,309 @@
+     * Retrieves a PIN for a given card
+     *
+     * @param cardId          the ID of the card (looks like ic_abcdef1234)
+     * @param verificationId  the ID of the verification that was sent to the cardholder
+     *                        (typically server-side, through /v1/issuing/verifications)
+     * @param userOneTimeCode the one-time code that was sent to the cardholder through sms or email
+     * @param listener        a listener for either the PIN, or any error that can occur
+     */
+    public void retrievePin(
+            @NonNull String cardId,
+            @NonNull String verificationId,
+            @NonNull String userOneTimeCode,
+            @Nullable IssuingCardPinRetrievalListener listener
+    ) {
+
+        Map<String, Object> arguments = new HashMap<>();
+        arguments.put(ARGUMENT_CARD_ID, cardId);
+        arguments.put(ARGUMENT_VERIFICATION_ID, verificationId);
+        arguments.put(ARGUMENT_ONE_TIME_CODE, userOneTimeCode);
+
+        final String operationId = UUID.randomUUID().toString();
+        mRetrievalListeners.put(operationId, listener);
+        mEphemeralKeyManager.retrieveEphemeralKey(operationId, PIN_RETRIEVE, arguments);
+    }
+
+    /**
+     * Retrieves a PIN for a given card
+     *
+     * @param cardId          the ID of the card (looks like ic_abcdef1234)
+     * @param newPin          the new desired PIN
+     * @param verificationId  the ID of the verification that was sent to the cardholder
+     *                        (typically server-side, through /v1/issuing/verifications)
+     * @param userOneTimeCode the one-time code that was sent to the cardholder through sms or email
+     * @param listener        a listener for either the PIN, or any error that can occur
+     */
+    public void updatePin(
+            @NonNull String cardId,
+            @NonNull String newPin,
+            @NonNull String verificationId,
+            @NonNull String userOneTimeCode,
+            @Nullable IssuingCardPinUpdateListener listener
+    ) {
+
+        Map<String, Object> arguments = new HashMap<>();
+        arguments.put(ARGUMENT_CARD_ID, cardId);
+        arguments.put(ARGUMENT_NEW_PIN, newPin);
+        arguments.put(ARGUMENT_VERIFICATION_ID, verificationId);
+        arguments.put(ARGUMENT_ONE_TIME_CODE, userOneTimeCode);
+
+        final String operationId = UUID.randomUUID().toString();
+        mUpdateListeners.put(operationId, listener);
+        mEphemeralKeyManager.retrieveEphemeralKey(operationId, PIN_UPDATE, arguments);
+    }
+
+    @Override
+    public void onKeyUpdate(@NonNull IssuingCardEphemeralKey ephemeralKey,
+                            @Nullable String operationId,
+                            @Nullable String action,
+                            @Nullable Map<String, Object> arguments) {
+
+        if (PIN_RETRIEVE.equals(action)) {
+            IssuingCardPinRetrievalListener listener = mRetrievalListeners.remove(operationId);
+            if (listener == null) {
```

Comment:
Should we mark the public methods as requiring a `@NonNull` listener?

Reply:
Yes